polytope_server/common/request_store/dynamodb_request_store.py

```
@@ -0,0 +1,243 @@
+        if ascending is not None and descending is not None:
+            raise ValueError("Cannot sort by ascending and descending at the same time.")
+
+        query = _make_query(**kwargs)
+        if user is not None:
+            key_cond_expr = Key("user_id").eq(str(user.id))
+            fn = self.table.query
+            params = {
+                "IndexName": "user-index",
+                "KeyConditionExpression": key_cond_expr,
+            }
+            if status is not None:
+                query["status"] = status.value
+        elif status is not None:
+            key_cond_expr = Key("status").eq(status.value)
+            fn = self.table.query
+            params = {
+                "IndexName": "status-index",
+                "KeyConditionExpression": key_cond_expr,
+            }
+        else:
+            fn = self.table.scan
+            params = {}
+
+        if query:
+            filter_expr = reduce(operator.__and__, (Attr(key).eq(value) for key, value in query.items()))
+            params["FilterExpression"] = filter_expr
+
+        if limit is not None:
+            params["Limit"] = limit
+
+        reqs = (_load(item) for item in _iter_items(fn, **params))
+        if ascending:
+            return sorted(reqs, key=lambda req: getattr(req, ascending))
+        if descending:
+            return sorted(reqs, key=lambda req: getattr(req, descending), reverse=True)
+        return list(reqs)
+
+    def update_request(self, request):
+        now = dt.datetime.now(dt.timezone.utc)
+        request.last_modified = now.timestamp()
+        self.table.put_item(Item=_dump(request))
+
+        if self.metric_store:
+            self.metric_store.add_metric(RequestStatusChange(request_id=request.id, status=request.status))
+
+        logger.info("Request ID %s status set to %s.", request.id, request.status)
+
+    def wipe(self):
+        pass  # Not needed for unit testing against moto
```

Should we raise an error in case someone try to use it?